gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/ddl/catalog/GfxdSystemProcedures.java

```diff
@@ -1717,7 +1717,16 @@ public static void CREATE_ALL_BUCKETS(String tableName) throws SQLException,
     try {
       final GemFireContainer container = CallbackProcedures
           .getContainerForTable(schema, table);
-      CREATE_ALL_BUCKETS_INTERNAL(container.getRegion(), tableName);
+      LocalRegion region = container.getRegion();
+      if (region.getAttributes().getPartitionAttributes() != null) {
+        if (((PartitionedRegion)region).
+            getRegionAdvisor().getCreatedBucketsCount() == 0) {
+          CREATE_ALL_BUCKETS_INTERNAL(container.getRegion(), tableName);
```

Comment:
Why should CREATE_ALL_BUCKETS be skipped if any bucket is already present? Caller should take care to invoke only if required.